envisage/ui/single_project/tests/test_clean_strings.py

```diff
@@ -0,0 +1,87 @@
+# -----------------------------------------------------------------------------
+#
+#  Copyright (c) 2019, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in /LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+# -----------------------------------------------------------------------------
+from __future__ import unicode_literals
+
+import unittest
+
+import six
+
+from envisage.ui.single_project.project import clean_filename
+
+# Safe strings should only contain the following characters.
+LEGAL_CHARS = set("-0123456789_abcdefghijklmnopqrstuvwxyz0123456789")
```

Comment:
Not sure why those digits are being repeated here. I need to fix that in Traits, but please could you fix in this PR, too?